mythril/laser/ethereum/strategy/custom.py

```diff
@@ -0,0 +1,55 @@
+from mythril.laser.ethereum.state.global_state import GlobalState
+from mythril.laser.ethereum.strategy.basic import BreadthFirstSearchStrategy
+from mythril.laser.ethereum import util
+from typing import Dict
+import logging
+
+
+JUMPDEST_LIMIT = 4
+log = logging.getLogger(__name__)
+
+
+class BFSBoundedLoopsStrategy(BreadthFirstSearchStrategy):
+    """Implements a breadth first search strategy that prunes loops.
+    JUMPI instructions are skipped after a jump destination has been
+    targeted JUMPDEST_LIMIT times.
+    """
+
+    def __init__(self, work_list, max_depth) -> None:
+        super().__init__(work_list, max_depth)
+        self._jumpdest_count = {}  # type: Dict[object, dict]
+
+    def get_strategic_global_state(self) -> GlobalState:
+        """
+        :return:
+        """
+
+        state = self.work_list.pop(0)
+
+        opcode = state.get_current_instruction()["opcode"]
+
+        if opcode == "JUMPI":
```

Comment:
We can reduce the nesting here by returning if the `opcode != "JUMPI"`, less nesting looks cool and easy to read.